### PR TITLE
remove tmp directories created by previous servers

### PR DESCRIPTION
**Pull request by BruceMacD**
with packing in cuda libs these start to get pretty big, clean temp `ollama-` dirs up before creating new ones.

### CODE REVIEW

File: server/routes.go [hunk 1]
```diff
@@ -586,3 +587,37 @@ func streamResponse(c *gin.Context, ch chan any) {
...
+	// calling these loads the runners into a temporary directory
+	llm.GGMLRunners()
+	llm.GGUFRunners()
+}
+
+func cleanupServer() {
+	if loaded.llm != nil {
+		loaded.llm.Close()
+	}
+	tmpDirs := []string{}
```
Comment (jmorganca) [inline server/routes.go:601]:
```suggestion
	var tmpDirs []string
```

File: server/routes.go [hunk 2]
```diff
@@ -586,3 +587,38 @@ func streamResponse(c *gin.Context, ch chan any) {
 		return true
 	})
 }
+
+func initServer() {
+	// calling these loads the runners into a temporary directory
+	llm.GGMLRunners()
+	llm.GGUFRunners()
```
Comment (mxyng) [inline server/routes.go:594]: Could be a separate PR. Setting up the runners shouldn't happen until they're needed.

File: server/routes.go [hunk 3]
```diff
@@ -586,3 +587,38 @@ func streamResponse(c *gin.Context, ch chan any) {
...
+}
+
+func cleanupServer() {
+	if loaded.llm != nil {
+		loaded.llm.Close()
+	}
+	var tmpDirs []string
+	if len(llm.GGMLRunners()) > 0 {
```
Comment (mxyng) [inline server/routes.go:602]: It might be easier to set up a top-level ollama working directory and store everything in there, e.g. `/tmp/ollamaXXXXXX/{ggml,gguf}/server`

**Review by jmorganca — APPROVED** [review]
